snooty/gizaparser/nodes.py

```diff
@@ -203,7 +205,9 @@ def add(self, path: Path, text: str, elements: Sequence[_I]) -> None:
 
             self.dg.add_edge(file_id, inherit.file)
 
-    def reify(self, obj: _I, diagnostics: List[Diagnostic]) -> _I:
+    def reify(
+        self, obj: _I, diagnostics: List[Diagnostic], refs_set: Set[Optional[str]]
```

Comment:
Why is the refs set a set of optional strings? I'm not sure it makes sense to put None into a set.